amethyst_renderer/src/sprite.rs

```diff
@@ -61,6 +68,141 @@ impl From<[f32; 4]> for Sprite {
     }
 }
 
+/// SystemData containing the data necessary to handle new rendered sprites
+#[derive(SystemData)]
+pub struct SpriteRenderData<'a> {
+    /// Storage containing the meshes
+    pub meshes: WriteStorage<'a, MeshHandle>,
+    /// Storage containing the materials
+    pub materials: WriteStorage<'a, Material>,
+    /// Loader resource
+    pub loader: ReadExpect<'a, Loader>,
+    /// Assets storage containing meshes
+    pub asset_storage: Read<'a, AssetStorage<Mesh>>,
+    /// Material defaults
+    pub material_defaults: ReadExpect<'a, MaterialDefaults>,
+}
+
+impl<'a> SpriteRenderData<'a> {
+    fn build_mesh_and_material(
+        &mut self,
+        sprite: &Sprite,
+        texture: TextureHandle,
+        size: (f32, f32),
+    ) -> (MeshHandle, Material) {
+        use vertex::PosTex;
+        let half_width = (sprite.right - sprite.left) * 0.5;
+        let half_height = (sprite.bottom - sprite.top) * 0.5;
+
+        let vertices: Vec<PosTex> = vec![
```

Comment:
We need to standardize this at some point. We could use genmesh or create our own crate containing those meshes that we re-use all the time.

Reply:
https://github.com/amethyst/amethyst/pull/777

Reply:
Ah it's been merged. Maybe I can rebase and use it then.